### DISCUSSION TITLE
Separate Workflow for compile-only to enable separate triggering of compile-only or full CI-run

### DISCUSSION THREAD

**Discussion by curateideas** [microsoft/AL-Go#118] [Ideas]
It would help developers to configure a workflow for compile only (like in Run-ALValidation) triggered by every push / pull request.

That would enable developer to create a new build and a compile result faster without the need to wait for feedback for the duration of a full CI-run. A full CI-run would be triggered by schedule.

Perhaps this scenario is already covered?

**Comment by freddydk** [comment]
The CI/CD workflow is triggered on push to main, release/* or feature/* branches + also for pull requests
Deploy part is only triggered on push to main.
The Test Current can be done on a schedule where you can test on multiple countries.

I don't believe in skipping tests to be able to check-in faster - IMO, main must always be stable.

Your idea does however make me think whether the feature branch CI pipeline could be a compile only - meaning that while working in a feature branch - no tests are running - that seems like a good idea and totally possible.
Does that sound useful?

**Reply by curateideas** [reply]
Totally useful for feature branches and for pull requests. 

I get your point that main must always be stable. We also contribute to that principle. But a little more scalability would be a time saver earlier in the process.

**Comment by StefanMaron** [comment]
Not sure if this idea was implemented already and if my idea is similar enough to revive this discussion.

Sometimes when doing quick changes it happens that the code I check in, does not even compile, because of typos, or perhaps because of code cops especially if fail-on-warning is enabled.
What do you guys think about a compile only step before the actual run pipeline step? I would think that this would not add much to the total run time of the pipeline but can save a huge amount of time, if the pipeline would fail on compile, since the container does not need to get created.
On top of this, the compile step, could also run on linux to further improve the run time.

And if this is a feature branch/PR the run-pipeline step (and windows runner) can be skipped entirely
